### DISCUSSION TITLE
There is a chromatic aberration between the origin picture and the texture which import an opengles texture

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#6035] [Q&A]
**Describe the bug**
I have closed the tone mapping !!!!!There is a chromatic aberration between the origin texture and imported texture. I create an opengles context outside and init a texture which is specified to rgba format by `glTexImage2D`. I create a `filament::Texture` to import the opengles texture and draw it on the screen. However, there is a chromatic aberration between the origin texture and imported texture. Please ignore the wrong ratio of the imported texture, it is just a demo. 

**To Reproduce**
1. create an openlges texture, and specify its format to `GL_RGBA`
2. decode this [png](https://drive.google.com/file/d/1KNPAXpAckzK79r9ATX04VkIP26kp5guC/view?usp=sharing) file and upload it to the opengles texture by using `texImage2D`
3. create a `filament::Texture` and import the opengles texture. Special its format to `Texture.InternalFormat.SRGB8_A8`
4. sample this filament::Texture` on a unlit mat file and  draw it on the screen
5. close tone mapping use this: `view.toneMapping = View.ToneMapping.LINEAR`

**initialize the opengles texture**
```
val textures = IntArray(1)
GLES30.glGenTextures(1, textures, 0)
val result = textures[0]
val textureTarget = if (external) GLES11Ext.GL_TEXTURE_EXTERNAL_OES else GLES30.GL_TEXTURE_2D
GLES30.glBindTexture(textureTarget, result)
GLES30.glTexImage2D(GLES30.GL_TEXTURE_2D, 0, GL_RGBA, CameraHelper.resolution.width, CameraHelper.resolution.height,
        0, GL_RGBA, GL_UNSIGNED_BYTE, null);
GLES30.glTexParameteri(textureTarget, GLES30.GL_TEXTURE_WRAP_S, GLES30.GL_CLAMP_TO_EDGE)
GLES30.glTexParameteri(textureTarget, GLES30.GL_TEXTURE_WRAP_T, GLES30.GL_CLAMP_TO_EDGE)
GLES30.glTexParameteri(textureTarget, GLES30.GL_TEXTURE_MIN_FILTER, GLES30.GL_NEAREST)
GLES30.glTexParameteri(textureTarget, GLES30.GL_TEXTURE_MAG_FILTER, GLES30.GL_NEAREST)
```

**initialize the filament texture**
```
filamentTexture = Texture.Builder()
.width(resolution.width)
.height(resolution.height)
.levels(1)
.sampler(Texture.Sampler.SAMPLER_2D)
.format(Texture.InternalFormat.SRGB8_A8)
.importTexture(externalTextureId.toLong())
.build(filamentEngine)
```

**mat**
```
material {
    // List of parameters exposed by this material
    parameters : [
        // The color must be passed in linear space, not sRGB
        {
            type : float3,
            name : baseColor
        },
        {
            type : sampler2d,
            name : videoTexture
        }
    ],
}

fragment {
    void material(inout MaterialInputs material) {
      vec4 color = texture(materialParams_videoTexture, uv);
      material.baseColor = color;
      material.baseColor.rgb = color.rgb;
     }.  
}

```

**Expected behavior**
no chromatic aberration 

**Screenshots**
the origin picture:
![image](https://user-images.githubusercontent.com/13013579/188159944-ac73a57c-3a68-4d11-ba1b-a6749bcd6d76.png)

the imported  texture:
<img width="391" alt="image" src="https://user-images.githubusercontent.com/13013579/188198365-ecf91464-5745-4e1e-b661-f52c9acbed62.png">

**Smartphone (please complete the following information):**
 - Device: [Samsung S20]
 - OS: [Android]

**Comment by romainguy** [comment]
Your texture is not set to `SRGBA8`

**Reply by dandingol03** [reply]
> Your texture is not set to `SRGBA8`

even i set the filament texture's format to `SRGB8_A8`, the rendered result still acts the same

**Reply by dandingol03** [reply]
> Your texture is not set to `SRGBA8`

your advice has no use

**Reply by romainguy** [reply]
Since it's an imported texture the sRGB format might be ignored by the driver. It's been a while since I've tried this combination. You could try to perform the conversion to linear space yourself by doing `material.baseColor.rgb = pow(material.baseColor, vec3(2.2));` at the end.

Btw are you just using an external texture to decode a PNG? Or doing something more complicated? If the former, you should consider not using external textures nor a separate GL context.
